qemu/tests/qemu_guest_agent.py

```diff
@@ -907,6 +907,36 @@ def gagent_check_fsfreeze(self, test, params, env):
 
         self._action_after_fsthaw()
 
+    @error_context.context_aware
+    def _change_bl(self, session):
+        """
+        Remove some qga cmds out of black list.
+        Now only linux guest has this behavior,but still leave interface
+        for windows guest.
+
+        :param session: VM session
+        """
+        if self.params.get("os_type") == "linux":
+            pattern = self.params["pattern"]
+            bl_check_cmd = self.params["black_list_check_cmd"] % pattern
+            bl_change_cmd = self.params["black_list_change_cmd"] % pattern
+            output = session.cmd_output(bl_check_cmd)
+            if output == "":
+                logging.info("%s related cmds are already "
+                             "in white list." % pattern)
+                return
+
+            session.cmd(bl_change_cmd)
```

Comment:
@xiagao If output already is ""(924 line is true), then don't need to execute this command. so, logical is wrong.  For 924  to 933 line , I think you can use one "if...else...".

Reply:
@xiangchunfu If output already is "", this function will return, it can not go to L929.